lnd_test.go

```diff
@@ -5320,29 +5322,33 @@ func assertActiveHtlcs(nodes []*lntest.HarnessNode, payHashes ...[]byte) error {
 			return fmt.Errorf("unable to get node chans: %v", err)
 		}
 
+		// Record all payment hashes active for this node.
+		htlcHashes := make(map[string]struct{})
```

Comment:
I still don't think this logic is correct. The new version will return true if _any_ of the active channels have this new HTLC. What we want to do is to return _false_ until _all_ channels for a particular node have the new HTLC.

It would seem that with the correction to the `listchannels` output, we don't need the second commit to modify the logic of `assertActiveHtlcs`.

Reply:
You're right, I was misinterpreting what the assertion was supposed to check. Pushed out a new version.

I still think the change to `assertActiveHtlcs` is warranted, as the old version wouldn't detect HTLCs with duplicaate `HashLock` (which this PR fixes), and also wouldn't detect if a channel only had a strict subset of `payHashes` active.